language/stdlib/modules/sliding_nonce.move

```diff
@@ -0,0 +1,45 @@
+address 0x0 {
+module SlidingNonce {
+    use 0x0::Transaction;
+
+    resource struct T {
+        min_nonce: u64,
+        nonce_mask: u128,
+    }
+
+    // Aborts transaction if nonce can not be accepted
+    public fun accept_nonce(seq_nonce: u64) acquires T {
```

Comment:
Nit: `advance_nonce` or some name that suggests that this mutates might be useful.